builtin/common/misc_helpers.lua

```diff
@@ -159,6 +159,56 @@ function dump(o, indent, nested, level)
 end
 
 --------------------------------------------------------------------------------
+-- Allow indexing strings
+
+getmetatable"".__index = function(str, index)
+	local itype = type(index)
+	if itype == "string" then
+		return string[index] or str:find(index)
```

Comment:
On the other hand, that might lead one mod to mess up the functionality of another by adding a function to string that clashes with a searched string of another mod.

Heisenbugs based on mod combinations.